File: sample/src/main/java/com/your_company/ui_bindings_sample/factories/CouponFactory.java
package com.your_company.ui_bindings_sample.factories;

import java.util.ArrayList;

import it.near.sdk.reactions.couponplugin.model.Claim;
import it.near.sdk.reactions.couponplugin.model.Coupon;

/**
 * @author Federico Boschini
 */

public class CouponFactory {

    private Coupon coupon = new Coupon();

    public CouponFactory() {
        Claim claim = new Claim();
        claim.serial_number = "0123456789";
        coupon.name = "Long name for a coupon";
        coupon.description = "Long coupon description, Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna ali.";
        coupon.value = "Value qwertyuioplkjhgfdsazxcvbnmpoiuytrewqasdfghj!";
        coupon.claims = new ArrayList<>();
        coupon.claims.add(claim);
    }

    public Coupon getValidCoupon() {
        coupon.redeemable_from = "2017-09-01T00:00:00.000Z";
        coupon.expires_at = "2099-10-30T23:59:59.999Z";
        return coupon;
    }

    public Coupon getInactiveCoupon() {
        coupon.redeemable_from = "2099-09-01T00:00:00.000Z";
        coupon.expires_at = "2099-10-30T23:59:59.999Z";
        return coupon;
    }

    public Coupon getExpiredCoupon() {
        coupon.redeemable_from = "2017-09-01T00:00:00.000Z";
        coupon.expires_at = "2017-09-02T23:59:59.999Z";
        return coupon;
    }

}

File: sample-kotlin/src/main/java/com/your_company/ui_bindings_kotlin_sample/factories/CouponFactory.kt
package com.your_company.ui_bindings_kotlin_sample.factories

import it.near.sdk.reactions.couponplugin.model.Claim
import it.near.sdk.reactions.couponplugin.model.Coupon

/**
* @author Federico Boschini
*/

object CouponFactory {

    private val coupon: Coupon = Coupon()

    init {
        val claim = Claim()
        claim.serial_number = "0123456789"
        @Suppress("DEPRECATION")
        coupon.name = "Long name for a coupon"
        coupon.description = "<a href=\"https://google.com\">LINK!</a>Long coupon description, Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna ali. https://nearit.com"
        coupon.value = "Value qwertyuioplkjhgfdsazxcvbnmpoiuytrewqasdfghj!"
        coupon.claims = listOf(claim)
    }

    fun getValidCoupon(): Coupon {
        coupon.redeemable_from = "2017-09-01T00:00:00.000Z"
        coupon.expires_at = "2099-10-30T23:59:59.999Z"
        return coupon
    }

    fun getInactiveCoupon(): Coupon {
        coupon.redeemable_from = "2099-09-01T00:00:00.000Z"
        coupon.expires_at = "2099-10-30T23:59:59.999Z"
        return coupon
    }

    fun getExpiredCoupon(): Coupon {
        coupon.redeemable_from = "2017-09-01T00:00:00.000Z"
        coupon.expires_at = "2017-09-02T23:59:59.999Z"
        return coupon
    }

}